### PR TITLE
[MRG+2] Document the documentation build requirements (closes #8174)

### LINKED ISSUE #8174: Update list of dependencies to build docs
The list at http://matplotlib.org/devdocs/devel/documenting_mpl.html#getting-started (sphinx+numpydoc) is outdated.  Compare with https://github.com/matplotlib/matplotlib/blob/master/doc/README.txt#L8 (sphinx numpydoc ipython mock colorspacious pillow).

### PR DESCRIPTION
**Pull request by vidursatija**
Only the file is edited and not build using Sphinx

### CODE REVIEW

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -8,8 +8,14 @@ Getting started
 ===============
 
 The documentation for matplotlib is generated from ReStructured Text using the
-Sphinx_ documentation generation tool. Sphinx-1.0 or later and numpydoc 0.4 or
-later is required.
+Sphinx_ documentation generation tool.
+The requirements are as follows (https://github.com/matplotlib/matplotlib/blob/master/doc-requirements.txt)
```
Comment (phobson) [inline doc/devel/documenting_mpl.rst]: I think you need a blank line before the start if the list for it to be properly formatted when it's rendered to HTML

Comment (vidursatija) [inline doc/devel/documenting_mpl.rst]: Sure. I'll do the needful

Comment (vidursatija) [inline doc/devel/documenting_mpl.rst]: I changed it. Now it looks like a proper list. Sorry for any inconvenience caused

**Comment by vidursatija** [comment]
Please pull the changes.

**Comment by phobson** [comment]
@vidursatija thanks for the changes and the PR in general. It's much appreciated. I'll try to find time to confirm that everything builds well tonight.

In the mean, perhaps it would make sense to embed the link to the requirements file. See [this reference](http://www.sphinx-doc.org/en/stable/rest.html#external-links).

Something like:

```
The `requirements`_ are as follows:

## yada yada yada

.. _requirements:  https://github.com/matplotlib/matplotlib/blob/master/doc-requirements.txt
```

**Comment by NelleV** [comment]
@phobson can you re-review?
@vidursatija Do you mind rebasing? I can help you out if you don't how to do it?

**Comment by vidursatija** [comment]
How can I rebase it?

**Comment by phobson** [comment]
@NelleV apporved!

**Comment by afvincent** [comment]
@vidursatija Here is some doc on [rebasing](http://matplotlib.org/devel/gitwash/development_workflow.html?highlight=rebasing#rebasing-a-pull-request-pr). I personally use the “interactive” fashion (`git rebase -i`), which (I find) gives you a bit more grip on what is done. I am not a heavy git user at all and it served me well up to now. If things go sour, you should find help on the [Gitter channel](https://gitter.im/matplotlib/matplotlib).

**Comment by vidursatija** [comment]
I think the problem occurred because I didn't change the branch while solving another PR. I'll still use rebase

**Comment by NelleV** [comment]
@vidursatija sorry it took so long to get back to you.

Though it is better to work in a branch, this is unfortunately not the issue here. Basically, someone else has been editing the same file as you, which means you'll have to update your code base. There are two ways to do this: (1) merging; (2) rebasing. The latter is more complicated (with IMO little to 0 benefit…), but I can walk you through it.

The first step is going to be to update your remote branches. Assuming your github remote is called "origin" and the main matplotlib one is called "matplotlib", you can do this by `git fetch matplotlib`. This will update the code in the remote branch `matplotlib/master`.

The second step is the rebasing itself. Rebasing will update your master branch in such a way that your commits come *after* the ones that have been merged into matplotlib/master.
To do this step, do `git rebase matplotlib/master`. You'll probably have a conflict in which case you'll have to edit the file with  the conflict to choose which lines to keep, and continue on with the rebase: first add the file(s) using `git add filename` to indicate that you've resolved the conflict, and then continue the rebase with `git rebase --continue` until the rebase is complete.

You can read more on rebase in our development guidelines: http://matplotlib.org/devel/gitwash/development_workflow.html#rebasing-a-pull-request-pr

**Comment by vidursatija** [comment]
I've resolved the conflicts by removing the conflicting lines.

**Comment by vidursatija** [comment]
Due to some mistake I deleted the forked repository. I still have to add changes. How can I do it?

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -17,6 +18,7 @@ The requirements are as follows (https://github.com/matplotlib/matplotlib/blob/m
 4. Mock
 5. colorspacious
 6. pillow
+=======
```
Comment (phobson) [inline doc/devel/documenting_mpl.rst]: i don't think this line of `====` belongs here

**Comment by NelleV** [comment]
@vidursatija Do you still have the local changes on your computer?

**Comment by vidursatija** [comment]
I have the file.

**Comment by NelleV** [comment]
Unfortunately, that won't be enough to retrieve the changes you've lost. There is unfortunately no other option to start from scratch.

**Comment by vidursatija** [comment]
Okay. I'll close this pull request and start a new one?

**Comment by NelleV** [comment]
closed in favor of #8373

**Pull request closed without merging**